node/core/provisioner/src/lib.rs

```diff
@@ -0,0 +1,362 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The provisioner is responsible for assembling a relay chain block
+//! from a set of available parachain candidates of its choice.
+
+use bitvec::vec::BitVec;
+use futures::{
+	channel::{mpsc, oneshot},
+	prelude::*,
+};
+use polkadot_node_subsystem::{
+	messages::{
+		AllMessages, ProvisionableData, ProvisionerInherentData, ProvisionerMessage,
+		RuntimeApiMessage,
+	},
+	util::{self, request_availability_cores, JobTrait, ToJobTrait},
+};
+use polkadot_primitives::v1::{BackedCandidate, CoreState, Hash, SignedAvailabilityBitfield};
+use std::{
+	collections::{HashMap, HashSet},
+	convert::TryFrom,
+	pin::Pin,
+};
+
+pub struct ProvisioningJob {
```

Comment:
Missing docs on public items, but I don't think this should be part of the public API of the crate anyway.